combat-trainer.lic

```diff
@@ -1424,16 +1425,97 @@ class AbilityProcess
 
     @barb_buffs = @buffs.delete('barb_buffs') || []
     echo("  @barb_buffs: #{@barb_buffs}") if $debug_mode_ct
+
     setup_barb_buff_flags
+
+    @paladin_use_badge = settings.paladin_use_badge
+    echo("  @paladin_use_badge: #{@paladin_use_badge}") if $debug_mode_ct
+
+    @paladin_glyphs = settings.paladin['Glyphs']
+    echo("  @paladin_glyphs: #{@paladin_glyphs}") if $debug_mode_ct
+
+    setup_glyph_flags unless @paladin_glyphs.empty?
+
+    return unless @paladin_use_badge
+
+    if DRCI.wearing?('pilgrim badge')
+      @paladin_badge_get_verb = 'remove'
+      @paladin_badge_store_verb = 'wear'
+    else
+      @paladin_badge_get_verb = 'get'
+      @paladin_badge_store_verb = 'stow'
+    end
+
+    UserVars.paladin_last_badge_use ||= Time.now
+    @badge_reuse_cooldown = 1860 # 31 minutes
   end
 
   def execute(game_state)
+    check_paladin_skills(game_state)
     check_nonspell_buffs(game_state)
     false
   end
 
   private
 
+  def check_paladin_skills(game_state)
+    return unless DRStats.paladin? 
+
+    pray_badge
+    check_glyphs
+  end
+
+  def setup_glyph_flags
+    @paladin_glyphs
+      .reject { |name|
```

Comment:
The block inside of `.reject` needs to return a bool. While this does, it's a bit unclear (I _think_ it's returning the `echo` result which ends up being true.

Reply:
Changed it, thanks, let me know if that is OK or if you know of a better method.

Reply:
Is the boolean here backwards? Since you're using `.reject`, a value of `false` means 'keep this item' and `true` means 'reject this item`.

If so, you could either swap `true` and `false` or change `.reject` to `.select`.

Reply:
ah yes, duh

Reply:
I'm going to change it to select, I think the reject and then unless read too difficult.